internal/venice-client-common/src/main/java/com/linkedin/venice/utils/RetryUtils.java

```diff
@@ -240,6 +240,31 @@ public static <T> T executeWithMaxAttemptAndExponentialBackoff(
     return Failsafe.with(retryPolicy).get(supplier::get);
   }
 
+  /**
+   * Execute a {@link Supplier} with maximum retries and fix duration per retry attempt. If all attempts are made and
+   * still no success, the last thrown exception will be thrown.
+   *
+   * @param supplier Execution unit which returns something ({@link T})
+   * @param maxRetry Total maximum retries made before giving up.
+   * @param durationPerAttempt Total duration per attempt. The delay after attempt will be the (duration per attempt - time spent in the current attempt).
+   * @param retryFailureTypes Types of failures upon which retry attempt is made. If a failure with type not specified
+   *                          in this list is thrown, it is thrown to the caller.
+   */
+  public static <T> T executeWithMaxRetriesAndFixedAttemptDuration(
+      VeniceCheckedSupplier<T> supplier,
+      final int maxRetry,
+      Duration durationPerAttempt,
+      List<Class<? extends Throwable>> retryFailureTypes) {
+    RetryPolicy<Object> retryPolicy = new RetryPolicy<>().handle(retryFailureTypes)
+        .withDelay(
+            (result, failure, context) -> (durationPerAttempt.toMillis() > context.getElapsedAttemptTime().toMillis()
+                ? Duration.ofMillis(durationPerAttempt.toMillis() - context.getElapsedAttemptTime().toMillis())
```

Comment:
Just curious: Is there anything here that prevents the actual task to be stuck forever? something like `maxDuration`.

Reply:
What is the issue we are facing?
`ContainsTopic` check stuck forever?
I guess there is no way for RetryUtil to control the actual runtime of task unless, the RetryUtil spawns a thread for every call and setup a timer against it.

Reply:
I don't think retryutil fully control the attempted task. Also, in our case it is containsTopic check, with succeed/failed very very fast.
The issue we are facing here is: We actually want to wait up to 180s for containsTopic check for Kafka metadata. However, since each check failed pretty fas and we were using exponential backoff retry strategy and started from 100ms, and it will be like:
check1 (some ms), wait 100ms, check 2 (some ms), wait 200ms, check 3 -> fail loudly within a second or so. 
I use the new API to make sure each attempt will be a full 60s and we retry 3 times as there is not point to aggressively query the Kafka.

Reply:
Thanks Lei's comment on the nit, will address it.

Reply:
Makes sense. Will mark as resolved.